renku/core/commands/migrate.py

```diff
@@ -45,38 +47,25 @@ def migrations_check():
 
 @inject.autoparams()
 def _migrations_check(client_dispatcher: IClientDispatcher):
+    """Check migration status of project."""
+
     client = client_dispatcher.current_client
 
-    template_update_possible, current_version, new_version = is_template_update_possible()
+    core_version, latest_version = _migrations_versions()
 
-    try:
-        template_source = client.project.template_source
-        template_ref = client.project.template_ref
-        template_id = client.project.template_id
-        automated_update = bool(client.project.automated_update)
-    except ValueError:
-        template_source = None
-        template_ref = None
-        template_id = None
-        automated_update = False
-
-    return (
-        is_migration_required(),
-        not is_project_unsupported(),
-        template_update_possible,
-        current_version,
-        new_version,
-        template_source,
-        template_ref,
-        template_id,
-        automated_update,
-        is_docker_update_possible(),
-    )
+    return {
+        "project_supported": not is_project_unsupported(),
+        "core_renku_version": core_version,
+        "project_renku_version": latest_version,
+        "core_compatibility_status": _metadata_migration_check(client),
+        "dockerfile_renku_status": _dockerfile_migration_check(client),
+        "template_status": _template_migration_check(client),
+    }
 
 
 def migrations_versions():
     """Return a command to get source and destination migration versions."""
-    return Command().command(_migrations_versions).lock_project().with_database()
```

Comment:
Isn't database required to read project's metadata?

Reply:
no, it falls back to reading `metadata.yml` otherwise migration wouldn't work.